src/components/MainStage/MainStageWrapper.tsx

```diff
@@ -218,7 +255,33 @@ const MainStageWrapper = () => {
     return { url: API_SOCKET_HOST, request }
   }
 
-  const onLeave = () => navigate(`/thankyou/${joinToken}`)
+  const getMenuActionsFromRole = (role: string) => {
+    const allowed = [UserRoles.ADMIN, UserRoles.ORGANIZER].map((r) => r.toLowerCase())
+    return allowed.indexOf(role) > -1 ? organizerSubscriberMenuActions : undefined
+  }
+
+  const onLeave = () => {
+    if (isAnonymous) {
+      navigate(`${Paths.ANONYMOUS_THANKYOU}/${joinToken}`)
+      return
+    }
+    navigate(`/thankyou/${joinToken}`)
+  }
+
+  const stopAnonJoinTimeout = () => {
+    if (anonJoinRef.current) {
+      clearTimeout(anonJoinRef.current)
+    }
+    clearTimeout(anonJoinTimeout)
+  }
+  const onAnonymousEntry = () => {
+    stopAnonJoinTimeout()
+    anonJoinTimeout = setTimeout(() => {
+      const { url, request } = getAnonymousSocketUrl(joinToken)
+      join(url, request)
+    }, 2000)
```

Comment:
why is this timeout needed here?

Reply:
Good catch. It was originally added to try and find where a continual hook re-render was occurring. I believe I found the reason and a proper solution and this was left around.

Will see if removing it still acts as expected.